asllib/builder.ml

```diff
@@ -86,27 +92,34 @@ let open_file filename =
   lexbuf_set_filename lexbuf filename;
   (lexbuf, chan)
 
-let from_file ?ast_type ?(allow_no_end_semicolon = false) version f =
+let from_file ?ast_type ?(allow_no_end_semicolon = false)
+    ?(allow_double_underscore = false) version f =
   let lexbuf, chan = open_file f in
   close_after chan
-  @@ from_lexbuf' ast_type allow_no_end_semicolon version lexbuf
+  @@ from_lexbuf' ast_type allow_no_end_semicolon allow_double_underscore
```

Comment:
I think we need a record type to pack together all the options instead of repeating them for every change in all of the places.

Reply:
I'll make it one record for the builder api. 

The original reason I didn't do that is because I wanted to keep the parser and lexer flags separate in two different records, but they would both have one element in them which seems a bit redundant for now.

Reply:
Added a record type. The code is a bit fishy in `aslref.ml` though. Turns out subsets of records don't play well with each other (this record ended up as a subset of the args record). I managed to get it to work without linter errors, by separating the two record references. Let me know if you know of a better way.